### PR TITLE
Addresses Issue #424

### PR DESCRIPTION
**Pull request by rajkiranrbala**
https://github.com/cloudfoundry-incubator/cf-abacus/issues/424

### CODE REVIEW

File: lib/utils/lock/src/index.js [hunk 1]
```diff
@@ -2,26 +2,33 @@
...
+  if(!locks[n])
+    locks[n] = new Lock();
+  const lock = locks[n];
+  const lockcb = (key, cb) => {
+    debug('Locking %s in %s locker', key, n);
+    lock(key, (release) => {
+      debug('Locked %s in %s locker', key, n);
+      cb(undefined, () => {
+        debug('Unlocking %s in %s locker', key, n);
+        release(() => {
```
Comment (hsiliev) [inline lib/utils/lock/src/index.js]: I hope I'm missing something, but we should remove from `locks` here, shouldn't we?

**Comment by maximilien** [comment]
Hi, @rajkiranrbala,

Any update on this PR. I think @hsiliev did provide a review. Let's move it forward it possible.

Thanks.

**Comment by rajkiranrbala** [comment]
I will have to make some changes. I will work on it.

**Comment by maximilien** [comment]
Thanks @rajkiranrbala. Looks like you might need a rebase

**Pull request merged**